tests/functional/contribute-signup.js

```diff
@@ -0,0 +1,171 @@
+        test.assertElementCount('.personal select', 1, 'Country select field exist');
+        test.assertElementCount('.personal textarea', 1, 'Message field exist');
+        test.assertElementCount('.form-format input[type="radio"]', 2, 'Email format radio buttons exist');
+        test.assertElementCount('.form-agree input[type="checkbox"]', 1, 'Accept privacy policy checkbox exists');
+        test.assertElementCount('.submit button', 1, 'Start Contributing buttons exists');
+        test.assertElementCount('.form-newsletter input[type="checkbox"]', 1, 'Newsletter subscribe checkbox exists');
+
+        test.assertVisible('.next-steps', 'The next steps section is visible');
+        test.assertElementCount('.steps li', 3, 'There are three steps listed');
+    });
+
+    casper.run(function() {
+        test.done();
+        helpers.done();
+    });
+});
+
+casper.test.begin('Contribute Signup, Areas of interest interaction: ' + url, 8, function suite(test) {
+    casper.start(url, function() {
+        casper.click('.option-list li:first-child label');
+    });
+
+    casper.waitUntilVisible('.areas', function() {
+        test.assert(true, 'The areas of interest section is visible');
+    });
+
+    casper.waitUntilVisible('#area-coding', function() {
+        test.assert(true, 'The appropriate area of interest select field is visible');
+        test.assertElementCount('#area-coding select', 1, 'The select field exists');
+
+        casper.click('.option-list li:nth-child(3) label');
+    })
+
+    casper.waitUntilVisible('#area-writing', function() {
+        test.assert(true, 'The writing area of interest section is visible');
+        test.assertElementCount('#area-writing select', 1, 'The select field exists');
+        test.assertNotVisible('#area-coding', 'The coding area of interest section has been hidden');
+
+        casper.click('.option-list li:nth-child(5) label');
+    });
+
+    casper.waitWhileVisible('#area-writing', function() {
+        test.assert(true, 'The writing area of interest is hidden');
+        test.assertNotVisible('.areas', 'Areas of interest select fields hidden');
+    });
+
+    casper.run(function() {
+        test.done();
+        helpers.done();
+    });
+});
+
+casper.test.begin('Contribute Signup, Tooltip: ' + url, 2, function suite(test) {
+    casper.start(url, function() {
+        this.mouse.move('a.info');
+    });
+
+    casper.waitUntilVisible('.tooltip', function() {
+        test.assert(true, 'The tooltip is shown');
+        // move the mouse of the info icon
+        this.mouse.move(0, 0);
+    });
+
+    casper.waitWhileVisible('.tooltip', function() {
+        test.assert(true, 'The tooltip has been removed');
+    });
+
+    casper.run(function() {
+        test.done();
+        helpers.done();
+    });
+});
+
+casper.test.begin('Contribute Sigup, Browser validation prevents form submission: ' + url, 1, function suite(test) {
```

Comment:
What do you think to losing this particular test? I don't think we strictly need to test native browser form validation works, especially given that we have a test below to make sure things still work correctly when a browser does not support said feature?